presto-hive/src/main/java/com/facebook/presto/hive/RetryDriver.java

```diff
@@ -43,6 +44,7 @@
     private final Function<Exception, Exception> exceptionMapper;
     private final List<Class<? extends Exception>> exceptionWhiteList;
     private final Optional<Runnable> retryRunnable;
+    private final List<Throwable> suppressedExceptions = new ArrayList<>();
```

Comment:
How does `suppressedExceptions` affect execution path? `suppressedExceptions` seems to be used only for writing.